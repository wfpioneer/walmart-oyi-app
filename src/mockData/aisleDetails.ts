import { AisleItem } from '../models/LocationItem';

export const mockAisles: AisleItem[] = [
  {
    aisleId: 1,
    aisleName: 'Aisle 1',
    sectionCount: 30
  },
  {
    aisleId: 2,
    aisleName: 'Aisle 2',
    sectionCount: 0
  },
  {
    aisleId: 3,
    aisleName: 'Aisle 3',
    sectionCount: 2
  },
  {
    aisleId: 4,
    aisleName: 'Aisle 4',
    sectionCount: 2
  },
  {
    aisleId: 5,
    aisleName: 'Aisle 5',
    sectionCount: 30
  },
  {
    aisleId: 6,
    aisleName: 'Aisle 6',
    sectionCount: 0
  }
];
