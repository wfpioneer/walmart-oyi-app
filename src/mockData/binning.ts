export const mockPallets = [{
  id: 123456,
  expirationDate: '10/3/2022',
  firstItem: {
    itemDesc: 'itemDesc',
    price: 123,
    upcNbr: '12343534',
    itemNbr: 351231,
    quantity: 2
  }
}, {
  id: 345345,
  expirationDate: '10/3/2022',
  firstItem: {
    itemDesc: 'itemDesc',
    price: 123,
    upcNbr: '345345',
    itemNbr: 351231,
    quantity: 2
  }
}];
