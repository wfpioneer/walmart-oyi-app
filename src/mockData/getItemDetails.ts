import ItemDetails from '../models/ItemDetails';
/* eslint-disable quote-props */

const floorType = 'Sales Floor';
const salesLastUpdateTs = '2020-07-15T08:02:17-05:00';
const salesDay1 = '2020-07-08';
const salesDay2 = '2020-07-09';
const salesDay3 = '2020-07-10';
const salesDay4 = '2020-07-11';
const salesDay5 = '2020-07-12';
const salesDay6 = '2020-07-13';
const salesDay7 = '2020-07-14';

export default {
  '123': {
    itemName: 'Test Item That is Really, Really Long (and has parenthesis)',
    itemNbr: 1234567890,
    upcNbr: '000055559999',
    status: 'Active',
    categoryNbr: 93,
    categoryDesc: 'Meat PI',
    price: 2000.94,
    basePrice: 1500.30,
    exceptionType: 'nsfl',
    imageUrlKey: undefined,
    imageBlobKey: undefined,
    completed: false,
    onHandsQty: 42,
    pendingOnHandsQty: -999,
    consolidatedOnHandQty: 10,
    claimsOnHandQty: 5,
    backroomQty: 6,
    cloudQty: undefined,
    replenishment: {
      onOrder: 48
    },
    location: {
      floor: [
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 1,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '1',
          locationName: 'A1-1',
          type: floorType,
          typeNbr: 8
        },
        {
          zoneId: 0,
          aisleId: 0,
          sectionId: 2,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '2',
          locationName: 'A1-2',
          type: 'End Cap',
          typeNbr: 12
        },
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 3,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '3',
          locationName: 'A1-3',
          type: 'Pod',
          typeNbr: 13
        },
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 4,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '4',
          locationName: 'A1-4',
          type: 'Display',
          typeNbr: 11
        }
      ],
      reserve: [
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 1,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '1',
          locationName: 'A1-1',
          type: 'Reserve',
          typeNbr: 7
        }
      ],
      count: 5
    },
    sales: {
      lastUpdateTs: salesLastUpdateTs,
      dailyAvgSales: 15,
      daily: [
        {
          day: salesDay1,
          value: 100
        },
        {
          day: salesDay2,
          value: 0
        },
        {
          day: salesDay3,
          value: 10
        },
        {
          day: salesDay4,
          value: 10
        },
        {
          day: salesDay5,
          value: 42
        },
        {
          day: salesDay6,
          value: 5
        },
        {
          day: salesDay7,
          value: 1
        }
      ],
      weeklyAvgSales: 10,
      weekly: [
        {
          week: 51,
          value: 39
        },
        {
          week: 52,
          value: 26
        },
        {
          week: 53,
          value: 45
        },
        {
          week: 54,
          value: 34
        },
        {
          week: 55,
          value: 34
        }
      ]
    }
  } as ItemDetails,
  '456': {
    itemName: 'Small, Store Use Item',
    itemNbr: 987654321,
    upcNbr: '777555333',
    status: 'Active',
    categoryNbr: 99,
    categoryDesc: 'Store Use',
    price: 2000.94,
    basePrice: 1500.30,
    exceptionType: 'nsfl',
    imageUrlKey: undefined,
    imageBlobKey: undefined,
    completed: false,
    onHandsQty: 42,
    pendingOnHandsQty: -999,
    consolidatedOnHandQty: 0,
    claimsOnHandQty: 5,
    backroomQty: 6,
    cloudQty: 42,
    replenishment: {
      onOrder: 48
    },
    location: {
      floor: [
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 1,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '1',
          locationName: 'A1-1',
          type: floorType,
          typeNbr: 8
        },
        {
          zoneId: 0,
          aisleId: 0,
          sectionId: 2,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '2',
          locationName: 'A1-2',
          type: 'End Cap',
          typeNbr: 12
        },
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 3,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '3',
          locationName: 'A1-3',
          type: 'Pod',
          typeNbr: 13
        },
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 4,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '4',
          locationName: 'A1-4',
          type: 'Display',
          typeNbr: 11
        }
      ],
      reserve: [
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 1,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '1',
          locationName: 'A1-1',
          type: 'Reserve',
          typeNbr: 7
        }
      ],
      count: 5
    },
    sales: {
      lastUpdateTs: salesLastUpdateTs,
      dailyAvgSales: 15,
      daily: [
        {
          day: salesDay1,
          value: 10
        },
        {
          day: salesDay2,
          value: 0
        },
        {
          day: salesDay3,
          value: 10
        },
        {
          day: salesDay4,
          value: 10
        },
        {
          day: salesDay5,
          value: 42
        },
        {
          day: salesDay6,
          value: 5
        },
        {
          day: salesDay7,
          value: 1
        }
      ],
      weeklyAvgSales: 10,
      weekly: [
        {
          week: 51,
          value: 0
        },
        {
          week: 1,
          value: 10
        },
        {
          week: 2,
          value: 27
        },
        {
          week: 3,
          value: 10
        }
      ]
    }
  } as ItemDetails,
  '789': {
    itemName: 'Wine Item',
    itemNbr: 987654321,
    upcNbr: '777555333',
    status: 'Active',
    categoryNbr: 19,
    categoryDesc: 'Wine',
    price: 2000.94,
    basePrice: 1500.30,
    exceptionType: 'nsfl',
    imageUrlKey: undefined,
    imageBlobKey: undefined,
    completed: false,
    onHandsQty: 42,
    consolidatedOnHandQty: 0,
    claimsOnHandQty: 0,
    backroomQty: 6,
    pendingOnHandsQty: 35,
    cloudQty: 42,
    replenishment: {
      onOrder: 48
    },
    location: {
      floor: [
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 1,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '1',
          locationName: 'A1-1',
          type: floorType,
          typeNbr: 8
        },
        {
          zoneId: 0,
          aisleId: 0,
          sectionId: 2,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '2',
          locationName: 'A1-2',
          type: 'End Cap',
          typeNbr: 12
        },
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 3,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '3',
          locationName: 'A1-3',
          type: 'Pod',
          typeNbr: 13
        },
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 4,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '4',
          locationName: 'A1-4',
          type: 'Display',
          typeNbr: 11
        }
      ],
      reserve: [
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 1,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '1',
          locationName: 'A1-1',
          type: 'Reserve',
          typeNbr: 7
        }
      ],
      count: 5
    },
    sales: {
      lastUpdateTs: salesLastUpdateTs,
      dailyAvgSales: 15,
      daily: [
        {
          day: salesDay1,
          value: 10
        },
        {
          day: salesDay2,
          value: 0
        },
        {
          day: salesDay3,
          value: 10
        },
        {
          day: salesDay4,
          value: 10
        },
        {
          day: salesDay5,
          value: 42
        },
        {
          day: salesDay6,
          value: 5
        },
        {
          day: salesDay7,
          value: 1
        }
      ],
      weeklyAvgSales: 10,
      weekly: [
        {
          week: 51,
          value: 0
        },
        {
          week: 1,
          value: 10
        },
        {
          week: 2,
          value: 27
        },
        {
          week: 3,
          value: 10
        }
      ]
    }
  } as ItemDetails
};
