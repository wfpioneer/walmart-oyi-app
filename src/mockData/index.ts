
import itemDetails from './getItemDetails';


export const getMockItemDetails = (upc: string) => {
  // @ts-ignore
  return itemDetails[upc] || {};
};
