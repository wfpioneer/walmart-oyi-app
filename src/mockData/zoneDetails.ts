
export interface DataSetOne {
    "id": number,
    "code": string,
    "category": string,
    "aisles": number
}


export const dataset_one : DataSetOne[] = [
    {
     "id": 1,
     "code": "B",
     "category": "B - Backroom",
     "aisles" : 4   
    },
    {
    "id": 2,
    "code": "C",
    "category": "C - Center",
    "aisles" : 2   
    },
    {
    "id": 3,
    "code": "E",
    "category": "Electronics",
    "aisles" : 6   
    },
    {
    "id": 4,
    "code": "FB",
    "category": "Fresh Bakery",
    "aisles" : 5  
    },
    {
    "id": 4,
    "code": "FC",
    "category": "Fresh Cafe",
    "aisles" : 4   
    },
    {
    "id": 5,
    "code": "FF",
    "category": "Fresh Freezer/Cooler",
    "aisles" : 2   
    },
    {
    "id": 6,
    "code": "FM",
    "category": "Fresh Meat, HMS & Deli",
    "aisles" : 6   
    },
    {
    "id": 7,
    "code": "FP",
    "category": "FB - Fresh Produce",
    "aisles" : 5  
    },
]
