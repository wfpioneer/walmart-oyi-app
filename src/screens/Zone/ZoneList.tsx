import React, {
  EffectCallback,
  useEffect,
  useMemo,
  useRef,
  useState
} from 'react';
import {
  ActivityIndicator, FlatList, Text, TouchableOpacity, View
} from 'react-native';
import MaterialCommunityIcon from 'react-native-vector-icons/MaterialCommunityIcons';
import { useDispatch } from 'react-redux';
import { Dispatch } from 'redux';
import {
  NavigationProp, RouteProp, useNavigation, useRoute
} from '@react-navigation/native';
import { BottomSheetModal, BottomSheetModalProvider } from '@gorhom/bottom-sheet';
import moment from 'moment';
import styles from './ZoneList.style';
import { useTypedSelector } from '../../state/reducers/RootReducer';
import LocationItemCard from '../../components/LocationItemCard/LocationItemCard';
import { strings } from '../../locales';
import { LocationHeader } from '../../components/locationHeader/LocationHeader';
import { getAllZones } from '../../state/actions/saga';
import { trackEvent } from '../../utils/AppCenterTool';
import { validateSession } from '../../utils/sessionTimeout';
import { AsyncState } from '../../models/AsyncState';
import COLOR from '../../themes/Color';
import { LocationType } from '../../models/LocationType';
import LocationManualScan from '../../components/LocationManualScan/LocationManualScan';
import { barcodeEmitter } from '../../utils/scannerUtils';
import { setManualScan, setScannedEvent } from '../../state/actions/Global';
import { hideLocationPopup } from '../../state/actions/Location';
import BottomSheetAddCard from '../../components/BottomSheetAddCard/BottomSheetAddCard';

const NoZonesMessage = () : JSX.Element => (
  <View style={styles.noZones}>
    <Text style={styles.noZonesText}>{strings('LOCATION.NO_ZONES_AVAILABLE')}</Text>
  </View>
);

interface ZoneProps {
    siteId: number,
    dispatch: Dispatch<any>,
    getZoneApi: AsyncState,
    isManualScanEnabled: boolean,
    navigation: NavigationProp<any>,
    apiStart: number,
    setApiStart: React.Dispatch<React.SetStateAction<number>>,
    route: RouteProp<any, string>,
    useEffectHook: (effect: EffectCallback, deps?:ReadonlyArray<any>) => void,
    trackEventCall: (eventName: string, params?: any) => void,
    locationPopupVisible: boolean
}

export const ZoneScreen = (props: ZoneProps) : JSX.Element => {
  const {
    siteId,
    getZoneApi,
    isManualScanEnabled,
    apiStart,
    setApiStart,
    dispatch,
    navigation,
    route,
    useEffectHook,
    trackEventCall,
    locationPopupVisible
  } = props;

  // calls the get all zone api
  useEffectHook(() => navigation.addListener('focus', () => {
    validateSession(navigation, route.name).then(() => {
      trackEventCall('get_zones_api_call');
      setApiStart(moment().valueOf());
      dispatch(getAllZones());
    }).catch(() => {});
  }), [navigation]);

  // scanned event listener
  useEffectHook(() => {
    const scanSubscription = barcodeEmitter.addListener('scanned', scan => {
      if (navigation.isFocused()) {
        validateSession(navigation, route.name).then(() => {
          trackEventCall('section_details_scan', { value: scan.value, type: scan.type });
          dispatch(setScannedEvent(scan));
          dispatch(setManualScan(false));
          navigation.navigate('SectionDetails');
        });
      }
    });
    return () => {
      scanSubscription.remove();
    };
  }, []);
  useEffectHook(() => {
    // on api success
    if (!getZoneApi.isWaiting && getZoneApi.result) {
      trackEventCall('get_zones_success', { duration: moment().valueOf() - apiStart });
    }

    // on api failure
    if (!getZoneApi.isWaiting && getZoneApi.error) {
      trackEventCall('get_zones_failure', {
        errorDetails: getZoneApi.error.message || getZoneApi.error,
        duration: moment().valueOf() - apiStart
      });
    }
  }, [getZoneApi]);

  if (getZoneApi.isWaiting) {
    return (
      <ActivityIndicator
        animating={getZoneApi.isWaiting}
        hidesWhenStopped
        color={COLOR.MAIN_THEME_COLOR}
        size="large"
        style={styles.activityIndicator}
      />
    );
  }

  if (getZoneApi.error) {
    return (
      <View style={styles.errorView}>
        <MaterialCommunityIcon name="alert" size={40} color={COLOR.RED_300} />
        <Text style={styles.errorText}>{strings('LOCATION.LOCATION_API_ERROR')}</Text>
        <TouchableOpacity
          style={styles.errorButton}
          onPress={() => {
            trackEventCall('location_api_retry',);
            dispatch(getAllZones());
          }}
        >
          <Text>{strings('GENERICS.RETRY')}</Text>
        </TouchableOpacity>
      </View>
    );
  }

  return (
    <View>
      {isManualScanEnabled && <LocationManualScan keyboardType="default" />}
      <LocationHeader
        location={`${strings('GENERICS.CLUB')} ${siteId}`}
        details={`${getZoneApi.result?.data.length || 0} ${strings('LOCATION.ZONES')}`}
      />
      <FlatList
        data={getZoneApi.result?.data || []}
        renderItem={({ item }) => (
          <LocationItemCard
            location={item.zoneName}
            locationId={item.zoneId}
            locationType={LocationType.ZONE}
            locationName={item.zoneName}
            locationDetails={`${item.aisleCount} ${strings('LOCATION.AISLES')}`}
            navigator={navigation}
            destinationScreen={LocationType.AISLE}
            dispatch={dispatch}
            locationPopupVisible={locationPopupVisible}
          />
        )}
        keyExtractor={item => item.zoneName}
        ListEmptyComponent={<NoZonesMessage />}
        contentContainerStyle={styles.contentPadding}
      />
    </View>
  );
};

const ZoneList = (): JSX.Element => {
  const { isManualScanEnabled } = useTypedSelector(state => state.Global);
  const siteId = useTypedSelector(state => state.User.siteId);
  const getZoneApi = useTypedSelector(state => state.async.getAllZones);
  const location = useTypedSelector(state => state.Location);
  const dispatch = useDispatch();
  const navigation = useNavigation();
  const [apiStart, setApiStart] = useState(0);
  const route = useRoute();

  const bottomSheetModalRef = useRef<BottomSheetModal>(null);

  const snapPoints = useMemo(() => ['22%', '50%'], []);

  useEffect(() => {
    if (navigation.isFocused() && bottomSheetModalRef.current) {
      if (location.locationPopupVisible) {
        bottomSheetModalRef.current.present();
      } else {
        bottomSheetModalRef.current.dismiss();
      }
    }
  }, [location]);

  return (
    <BottomSheetModalProvider>
      <TouchableOpacity
        onPress={() => dispatch(hideLocationPopup())}
        activeOpacity={1}
        disabled={!location.locationPopupVisible}
        style={styles.container}
      >
        <ZoneScreen
          siteId={siteId}
          dispatch={dispatch}
          getZoneApi={getZoneApi}
          isManualScanEnabled={isManualScanEnabled}
          navigation={navigation}
          route={route}
          useEffectHook={useEffect}
          apiStart={apiStart}
          setApiStart={setApiStart}
          trackEventCall={trackEvent}
          locationPopupVisible={location.locationPopupVisible}
        />
      </TouchableOpacity>
      <BottomSheetModal
        ref={bottomSheetModalRef}
        snapPoints={snapPoints}
        index={0}
        onDismiss={() => dispatch(hideLocationPopup())}
        style={styles.bottomSheetModal}
      >
        <BottomSheetAddCard
          isManagerOption={true}
          isVisible={true}
          text={strings('LOCATION.ADD_AREA')}
          onPress={() => {}}
        />
      </BottomSheetModal>
    </BottomSheetModalProvider>
  );
};

export default ZoneList;
