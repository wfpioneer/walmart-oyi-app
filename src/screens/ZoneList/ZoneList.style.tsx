import React from 'react';
import { StyleSheet, StatusBar } from 'react-native';


const styles = StyleSheet.create({
    container: {
      flex: 1,
      marginTop: StatusBar.currentHeight || 0,
    },
    item: {
      backgroundColor: '#fcfcfc',
      padding: 14,
      marginVertical: 1,
      marginHorizontal: 0,
    },
    staticHeader: {
        backgroundColor: '#fcfcfc',
        padding: 14,
        marginHorizontal: 0,
        zIndex: 2,
        borderBottomWidth: 2,
        borderBottomColor: '#dbdbdb'
      },
      aisleText: {
        color: '#b5b5b5',
        fontSize: 12,
      },
    title: {
      fontSize: 40,
    },
    itemContainer: {
    flex: 1,
    flexDirection: "row",
    alignItems: 'center'
    }
  });
  
  export default styles