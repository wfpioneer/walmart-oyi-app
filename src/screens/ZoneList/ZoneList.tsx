import React from 'react';
import { SafeAreaView, View, FlatList, Text, TouchableOpacity} from 'react-native';
import itemData from "../../mockData/getItemDetails";
import styles from './ZoneList.style' 
import MaterialCommunityIcon from 'react-native-vector-icons/MaterialCommunityIcons';
import { useTypedSelector } from '../../state/reducers/RootReducer';
import { Provider } from 'react-redux';
import {dataset_one, DataSetOne} from "../../mockData/zoneDetails";




const ItemCard = (item: DataSetOne) => {
    return (
        <TouchableOpacity style={styles.item}>
            <View style={styles.itemContainer}>
                <View style={{flex:8}}> 
                     <Text>{item.code} - {item.category}</Text>
                    <View>
                    <Text style={styles.aisleText}> {item.aisles} aisles</Text>
                    </View>
               </View>
               <View style={{flex:1}}>  
                    <MaterialCommunityIcon name="greater-than" size={18} color="#349beb"/>
               </View>
            </View>
        </TouchableOpacity>
        )
} 

const ZoneList = () => {
    const siteId = useTypedSelector(state => state.User.siteId)
    return (<ZoneListScreen dataSet={dataset_one} siteId = {siteId}/>)
}



interface ZoneListProps {
    siteId: number,
    dataSet: DataSetOne[]
}

export const ZoneListScreen = (props: ZoneListProps) => {
    const siteId = props.siteId;
    const data_set = props.dataSet;

    return (
        <>
        <View style={styles.staticHeader}>
        <Text>CLUB {siteId}</Text>
        </View>
          <FlatList
            data={data_set}
            renderItem={({item}) => {
                return ItemCard(item)
            }}
            keyExtractor={item => item.category}/>
        </>
      );

}

export default ZoneList;
