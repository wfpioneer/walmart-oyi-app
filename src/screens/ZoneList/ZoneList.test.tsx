import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import {ZoneListScreen} from './ZoneList';
import {dataset_one} from "../../mockData/zoneDetails";

describe("Zone List", () => {
    it('Renders Zone Screen', () => {
        const renderer = ShallowRenderer.createRenderer();
        renderer.render(<ZoneListScreen dataSet={dataset_one} siteId={5522}/>)
        expect(renderer.getRenderOutput()).toMatchSnapshot();
      });


})

