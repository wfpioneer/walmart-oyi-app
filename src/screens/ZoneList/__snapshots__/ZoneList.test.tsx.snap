// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Zone List Renders Zone Screen 1`] = `
<React.Fragment>
  <View
    style={
      Object {
        "backgroundColor": "#fcfcfc",
        "borderBottomColor": "#dbdbdb",
        "borderBottomWidth": 2,
        "marginHorizontal": 0,
        "padding": 14,
        "zIndex": 2,
      }
    }
  >
    <Text>
      CLUB 
      5522
    </Text>
  </View>
  <FlatList
    data={
      Array [
        Object {
          "aisles": 4,
          "category": "B - Backroom",
          "code": "B",
          "id": 1,
        },
        Object {
          "aisles": 2,
          "category": "C - Center",
          "code": "C",
          "id": 2,
        },
        Object {
          "aisles": 6,
          "category": "Electronics",
          "code": "E",
          "id": 3,
        },
        Object {
          "aisles": 5,
          "category": "Fresh Bakery",
          "code": "FB",
          "id": 4,
        },
        Object {
          "aisles": 4,
          "category": "Fresh Cafe",
          "code": "FC",
          "id": 4,
        },
        Object {
          "aisles": 2,
          "category": "Fresh Freezer/Cooler",
          "code": "FF",
          "id": 5,
        },
        Object {
          "aisles": 6,
          "category": "Fresh Meat, HMS & Deli",
          "code": "FM",
          "id": 6,
        },
        Object {
          "aisles": 5,
          "category": "FB - Fresh Produce",
          "code": "FP",
          "id": 7,
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</React.Fragment>
`;
